### PR TITLE
[BugFix] Fix to prevent resource leaks in 3 classes 

### PR DESCRIPTION
**Pull request by dk2k**
Implemented try-with-resources for instances of Autocloseable

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLoggerAllocator.java [hunk 1]
```diff
@@ -180,23 +180,24 @@ private synchronized BufferedLogChannel allocateNewLog(File dirForNextEntryLog,
         } while (testLogFile == null);
 
         File newLogFile = new File(dirForNextEntryLog, logFileName);
-        FileChannel channel = new RandomAccessFile(newLogFile, "rw").getChannel();
+        try (FileChannel channel = new RandomAccessFile(newLogFile, "rw").getChannel()) {
```
Comment (dlg99) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLoggerAllocator.java:183]: is it a good idea to close the channel here? it is used by the logChannel that is being returned

Comment (dk2k) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLoggerAllocator.java]: Reverted changes for this file

**Review by dlg99 — APPROVED** [review]

**Review by shoothzj — COMMENTED** [review]
It seems we have been leaked some resources for long, really? Need more eyes. @nicoloboschi @hangc0276 @nodece

**Review by nodece — APPROVED** [review]

**Comment by nodece** [comment]
> It seems we have been leaked some resources for long, really? Need more eyes. @nicoloboschi @hangc0276 @nodece

Looks like it. We should take care of it.

**Review by shoothzj — APPROVED** [review]

**Review by shoothzj — CHANGES_REQUESTED** [review]
LGTM, but I think we need to change the commit and PR title. It's a fix of leak

**Review by fu-turer — APPROVED** [review]

**Review by nicoloboschi — APPROVED** [review]

**Review by shoothzj — APPROVED** [review]

**Comment by shoothzj** [comment]
Thanks for your contribution!

**Pull request merged**